dotnet/src/SemanticKernel.Core/Planning/Plan.cs

```diff
@@ -292,28 +291,19 @@ protected override SKFunctionMetadata GetMetadataCore()
         }
         else
         {
-            var invokingEventArgs = this.CallFunctionInvoking(kernel, context);
-            if (invokingEventArgs.IsSkipRequested || invokingEventArgs.CancelToken.IsCancellationRequested)
-            {
-                return new FunctionResult(this.Name, context)
-                {
-                    IsCancellationRequested = invokingEventArgs.CancelToken.IsCancellationRequested,
-                    IsSkipRequested = invokingEventArgs.IsSkipRequested
-                };
-            }
-
             // loop through steps and execute until completion
             while (this.HasNextStep)
             {
                 AddVariablesToContext(this.State, context);
 
                 var stepResult = await this.InternalInvokeNextStepAsync(kernel, context, cancellationToken).ConfigureAwait(false);
 
-                if (stepResult.IsCancellationRequested)
+                // If a step was cancelled before invocation
+                // Return the last result state of the plan.
+                if (stepResult is null || stepResult.IsCancellationRequested)
```

Comment:
What is the scenario where stepResult is null?

Reply:
This is no longer needed will remove it